internal/offline/check.go

```diff
@@ -0,0 +1,138 @@
+package offline
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/models"
+	"github.com/google/osv-scanner/pkg/osv"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+const zippedDBRemoteHost = "https://osv-vulnerabilities.storage.googleapis.com"
+const envKeyLocalDBCacheDirectory = "OSV_SCANNER_LOCAL_DB_CACHE_DIRECTORY"
+
+func loadDB(dbBasePath string, ecosystem lockfile.Ecosystem, offline bool) (*ZipDB, error) {
+	return NewZippedDB(dbBasePath, string(ecosystem), fmt.Sprintf("%s/%s/all.zip", zippedDBRemoteHost, ecosystem), offline)
+}
+
+func toPackageDetails(query *osv.Query) (lockfile.PackageDetails, error) {
+	if query.Package.PURL != "" {
+		pkg, err := models.PURLToPackage(query.Package.PURL)
+
+		if err != nil {
+			return lockfile.PackageDetails{}, err
+		}
+
+		return lockfile.PackageDetails{
+			Name:      pkg.Name,
+			Version:   pkg.Version,
+			Ecosystem: lockfile.Ecosystem(pkg.Ecosystem),
+			CompareAs: lockfile.Ecosystem(pkg.Ecosystem),
+		}, nil
+	}
+
+	return lockfile.PackageDetails{
+		Name:      query.Package.Name,
+		Version:   query.Version,
+		Commit:    query.Commit,
+		Ecosystem: query.Package.Ecosystem,
+		CompareAs: query.Package.Ecosystem,
+	}, nil
+}
+
+func setupLocalDBDirectory(localDBPath string) (string, error) {
```

Comment:
Can you please add more comments inside this function to enumerate all the different cases that are being handled?

Reply:
I ended up heavily refactoring the internals of this function, which included catching a couple of bugs